lib/assert.js

```diff
@@ -280,9 +282,84 @@ function _deepEqual(actual, expected, strict, memos) {
   memos.actual.push(actual);
   memos.expected.push(expected);
 
+
   return objEquiv(actual, expected, strict, memos);
 }
 
+function setEquiv(a, b, strict, actualVisitedObjects) {
+  // This code currently returns false for this pair of sets:
+  //   assert.deepEqual(new Set(['1', 1]), new Set([1]))
+  //
+  // In theory, all the items in the first set have a corresponding == value in
+  // the second set, but the sets have different sizes. Its a silly case,
+  // and more evidence that deepStrictEqual should always be preferred over
+  // deepEqual. The implementation currently returns false, which is simpler
+  // and slightly faster.
+  if (a.size !== b.size)
+    return false;
+
+  outer: for (const val1 of a) {
+    if (!b.has(val1)) {
+      // The value doesn't exist in the second set by reference, so we'll go
+      // hunting for something thats deep-equal to it. Note that this is O(n^2)
+      // complexity, and will get slower if large, very similar sets / maps are
+      // nested inside. Unfortunately there's no real way around this.
```

Comment:
Yes. Thats really clever, and very obvious now that you've pointed it out. Fixed.